packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/common/useMemberData.ts

```diff
@@ -0,0 +1,63 @@
+import app from 'state';
+import { useFetchGroupsQuery } from 'state/api/groups';
+import { Memberships } from 'state/api/groups/refreshMembership';
+import { CWTableState } from 'views/components/component_kit/new_designs/CWTable/useCWTableState';
+import useGetMembersQuery from '../../../../state/api/members/getMembers';
+
+interface UseMemberDataProps {
+  memberships?: Memberships[];
+  tableState?: CWTableState;
+  groupFilter?: string;
+  debouncedSearchTerm?: string;
+  membersPerPage: number;
+  page: number;
+  allowedAddresses: string[];
+}
+
+export const useMemberData = ({
+  memberships,
+  tableState,
+  groupFilter,
+  debouncedSearchTerm,
+  membersPerPage,
+  page,
+  allowedAddresses,
+}: UseMemberDataProps) => {
+  const parseMembership = !isNaN(groupFilter as unknown as number)
+    ? `in-group:${groupFilter}`
+    : groupFilter;
+
+  const membershipsFilter = ['all-community'].includes(groupFilter.toString())
+    ? undefined
+    : parseMembership;
+
+  const { data: members, isLoading: isLoadingMembers } = useGetMembersQuery({
+    limit: membersPerPage,
+    order_by: tableState.orderBy,
+    order_direction: tableState.orderDirection as 'ASC' | 'DESC',
```

Comment:
i think we are missing a pipe (`|`), single `|` would do bitwise OR

Reply:
But it is type information (The `as` keyword). This makes it a discriminated union, not a bitwise operation.

Reply:
ah i see, mb.